### ISSUE TITLE
Chinese encoding problem when using mdmath.clipToHtml

### ISSUE THREAD

**Issue by TonySFU** [goessner/mdmath#13]
### Summary
When I use mdmath.clipToHtml, I find out there is an encoding problem after I paste into an HTML file, if I use mdmath.exportToHtml, and copy from output, it works fine.

### Expected behavior
mdmath.clipToHtml works same as mdmath.exportToHtml

### Actual behavior
Using mdmath.clipToHtml can not get the correct encoding output.

### Related links

* [Markdown+Math](https://github.com/goessner/mdmath)

### Environment

```text
     Operating system: macOS 10.12.3
       VSCode version: 1.9.1
Markdown+Math version: 1.2.0
```

**Comment by goessner** [comment]
Hmm .. I tested it with another asian file. It obviously works fine. The Html templates for both commands are identical.

Can you provide me a short test file for replicating the encoding problem ?

Thanks

**Comment by TonySFU** [comment]
I did some tests on different platforms.
test file just use this below:

```
# 一级标题
* HI
* HIHI
## 二级标题
### 三级标题
```

everything works great on my Windows 10 desktop. System language Chinese, vs code language both English and Chinese.

Problems show up on my MacBook Pro, 10.12.3

System language English, vs code language English:
mdmath.exportToHtml shows HTML in OUTPUT works fine,
mdmath.clipToHtml has encoding problem.

System language Chinese, vs code language English:
mdmath.exportToHtml shows HTML in OUTPUT works fine,
mdmath.clipToHtml not work, can not get anything from system clipboard, command + V not work.

It looks like some system clipboard API problem.

I am a student want to use Markdown and Math to write my homework. The final product will be the .pdf, for now I can convert HTML to PDF. I will appreciate if you can add export PDF in future version, thank you for your time.

**Comment by TonySFU** [comment]
I find this problem in the code, the 'copy-paste' npm can not get the right output.

I did test by use this code in node.js
`var ncp = require("copy-paste");
ncp.copy('一级标题', function() {
    console.log("Finished")
})`

it gave me this `‰∏ÄÁ∫ßÊ†áÈ¢ò` is same as what happened in your extension.

**Comment by floatdrop** [comment]
@TonySFU can you check, that [clipboardy](http://npmjs.com/clipboardy) works for you? If so `copy-paste` can be replaced.

**Comment by goessner** [comment]
I will try it out .. but it has to wait until end of next week.

thanx for your effort.

**Comment by TonySFU** [comment]
I found clipboardy works great on system Node environment, but not works under VScode(inside) Node  environment.
When I run this code 
<img width="867" alt="screen shot 2017-05-11 at 7 15 37 pm" src="https://cloud.githubusercontent.com/assets/22601843/25979588/46637714-367e-11e7-9f0f-2f46288184a9.png">

by using system node.js command, it return the correct result.
But when I set a DEBUG under VScode (set up a lanuch.json file) with this code, it gave me this result.
<img width="864" alt="screen shot 2017-05-11 at 7 16 35 pm" src="https://cloud.githubusercontent.com/assets/22601843/25979607/688bdf70-367e-11e7-82f0-f4df298a5863.png">

which is not correct encoding. 
So I think the problem is inside the vscode(Node) between system clipboard(macOS).
Just switch to another npm package NOT solve this issue, even if it works under macOS system node.js.

**Comment by TonySFU** [comment]
I opened an issue on vscode repository.
https://github.com/Microsoft/vscode/issues/26502

**Comment by weinand** [comment]
This is not a VS Code issue: "clipboardy" relies on a "LC_CTYPE" environment variable which is not set when an macOS application is launched from the Dock or by double clicking in the finder.
We won't fix this in VS Code because that fix would only work while developing an application that uses "clipboardy" inside VS Code. If this application is deployed (outside of VS Code) the fix would no longer be available and the application would fail. So it is the responsibility of the application to deal with the missing "LC_CTYPE" environment variable.

For details see https://github.com/Microsoft/vscode/issues/26502#issuecomment-301043547

**Comment by goessner** [comment]
Thanks a lot for fixing this issue and recommending `clipboardy`.

@floatdrop: I worked your suggested modifications with some other changes into the source.

@TonySFU, @floatdrop: I tried to find a way to add you both as contributors without explicite pull requests. I didn't find anything. So I added you as contributors in Readme file. (Maybe you know here a solution).

I took your modifications from source without being able to test it, as I don't own a mac book .
